cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -128,14 +128,29 @@ func SetAPIEndpointDynamicDefaults(cfg *kubeadmapi.APIEndpoint) error {
 
 // SetClusterDynamicDefaults checks and sets values for the ClusterConfiguration object
 func SetClusterDynamicDefaults(cfg *kubeadmapi.ClusterConfiguration, advertiseAddress string, bindPort int32) error {
+
+	kubeProxyUserBindAddress := ""
+	if cfg.ComponentConfigs.KubeProxy != nil && cfg.ComponentConfigs.KubeProxy.BindAddress != "" {
+		kubeProxyUserBindAddress = cfg.ComponentConfigs.KubeProxy.BindAddress
+	}
+
 	// Default all the embedded ComponentConfig structs
+	// This will set `cfg.ComponentConfigs.KubeProxy.BindAddress` as `0.0.0.0` if it is unset
 	componentconfigs.Known.Default(cfg)
 
-	ip := net.ParseIP(advertiseAddress)
-	if ip.To4() != nil {
-		cfg.ComponentConfigs.KubeProxy.BindAddress = kubeadmapiv1beta2.DefaultProxyBindAddressv4
+	// validate cfg.ComponentConfigs.KubeProxy.BindAddress.
+	if kubeProxyUserBindAddress != "" {
+		addressIP := net.ParseIP(kubeProxyUserBindAddress)
```

Comment:
Since kube-proxy is a daemonset, so I think the bind address only has two options: either `0.0.0.0` or `::`.

Setting it to a specific IP address will make it bind fail on other nodes since all nodes have same configuration.